File: LeetCode/Easy/Problems 151 - 200/Problem 189/p189Sol2.c
void rotate(int* nums, int numsSize, int k){
    int* numsOriginal = malloc(numsSize * sizeof(int));
    for (int i = 0; i < numsSize; i++) {
        numsOriginal[i] = nums[i];
    }
    
    for (int i = 0; i < numsSize; i++) {
        nums[(i+k)%numsSize] = numsOriginal[i];    
    }
    
    free(numsOriginal);
}

File: LeetCode/Easy/Problems 151 - 200/Problem 189/p189Sol2.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        numsOriginal = [x for x in nums] 
        
        for i in range(length):
            nums[(i+k)%length] = numsOriginal[i]